File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/C/Exercicio 21.c
/*
21 - Elabore um programa que receba a matrícula, a nota 1 e nota 2 de uma lista de alunos e
escreva essas informações em um arquivo (nome do arquivo é exercicioArquivo.dat). A
leitura de dados deve ser interrompida quando for digitado a matricula = 0.

*/

#include<stdio.h>
#include<stdlib.h>

int main ()
{
    int mat, i;
    float nota1, nota2;
    FILE *arq;
    arq = fopen("exercicioArquivo.dat","w+");

    if (arq == NULL) {
        printf("\nArquivo inválido");
        exit(1);
    }
    else {
        printf("Digite a mat:");
        scanf("%d", &mat);

        while (mat!=0){
            printf("\nDigite a N1:");
            scanf("%f", &nota1);
            printf("\nDigite a N2:");
            scanf("%f", &nota2);
            fprintf(arq, "%d %.2f %.2f\n", mat, nota1, nota2);
            printf("\nDigite a mat:");
            scanf("%d", &mat);
        }
        i= 1;
        rewind(arq);

        fscanf(arq, "%d %f %f", &mat, &nota1, &nota2);
        while (!feof(arq)) {
            printf("\nMatricula do aluno %d: %d - Nota1: %.2f / Nota2: %.2f", i++, mat, nota1, nota2);
            fscanf(arq, "%d %f %f", &mat, &nota1, &nota2);
        }
    }
    printf("\n\n");
    fclose(arq);
    system("pause");
 }

File: 01-FUNDAMENTOS_DE_PROGRAMACAO/Python/exercicio21.py
"""
21 - Elabore um programa que receba a matrícula, a nota 1 e nota 2 de uma lista de alunos e
escreva essas informações em um arquivo (nome do arquivo é arquivo.txt). A
leitura de dados deve ser interrompida quando for digitado a matricula = 0.
"""

f = open("arquivo.txt", "w")
mat = int(input('Digite a matrícula: '))
while(mat!=0):
    n1 = float(input('Digite a Nota 1: '))
    n2 = float(input('Digite a Nota 2: '))
    f.write(f'\n Matricula: '+str(mat))
    f.write(f'\n Nota 1: '+str(n1))
    f.write(f'\n Nota 2: '+str(n2))
    f.write('\n')
    mat = int(input('\nDigite a matrícula: '))
f.close()

#open and read the file after the appending:
f = open("arquivo.txt", "r")
print(f.read())